packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx

```diff
@@ -1,7 +1,174 @@
-import React from 'react';
+import { AccessLevel } from '@hicommonwealth/core';
+import axios from 'axios';
+import React, { useEffect, useMemo, useState } from 'react';
+import app from 'state';
+import { useDebounce } from 'usehooks-ts';
+import FeatureHint from 'views/components/FeatureHint';
+import { CWText } from 'views/components/component_kit/cw_text';
+import {
+  APIOrderBy,
+  APIOrderDirection,
+} from '../../../../../scripts/helpers/constants';
+import useSearchProfilesQuery from '../../../../../scripts/state/api/profiles/searchProfiles';
+import RoleInfo from '../../../../models/RoleInfo';
+import { ComponentType } from '../../../components/component_kit/types';
+import { UpgradeRolesForm } from '../../../pages/manage_community/upgrade_roles_form';
+import { ManageRoles } from '../../manage_community/manage_roles';
+import './AdminsAndModerators.scss';
 
 const AdminsAndModerators = () => {
-  return <div className="AdminsAndModerators">AdminsAndModerators</div>;
+  const [admins, setAdmins] = useState([]);
+  const [mods, setMods] = useState([]);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const debouncedSearchTerm = useDebounce<string>(searchTerm, 500);
+
+  const fetchAdmins = async () => {
+    const memberAdmins = [];
+    const memberMods = [];
+
+    try {
+      const res = await axios.get(`${app.serverUrl()}/roles`, {
+        params: {
+          chain_id: app.activeChainId(),
+          permissions: ['moderator', 'admin'],
+        },
+      });
+      const roles = res.data.result || [];
+      roles.forEach((role) => {
+        if (role.permission === AccessLevel.Admin) {
+          memberAdmins.push(role);
+        } else if (role.permission === AccessLevel.Moderator) {
+          memberMods.push(role);
+        }
+      });
```

Comment:
this request is also used in 3 other places, let's add a react query fetch (useQuery) for this.